test/extensions/filters/http/wasm/test_data/test_grpc_stream_cpp.cc

```diff
@@ -91,4 +92,32 @@ FilterHeadersStatus GrpcStreamContext::onRequestHeaders(uint32_t, bool) {
   return FilterHeadersStatus::StopIteration;
 }
 
+class GrpcStreamContext : public Context {
+public:
+  explicit GrpcStreamContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t, bool) override;
+};
+
+static RegisterContextFactory register_GrpcStreamContext(CONTEXT_FACTORY(GrpcStreamContext),
+                                                         ROOT_FACTORY(GrpcStreamRootContext),
+                                                         "grpc_stream");
+
+FilterHeadersStatus GrpcStreamContext::onRequestHeaders(uint32_t, bool) {
+  HeaderStringPairs initial_metadata;
+  if (root()->grpcStreamHandler("bogus service string", "service", "method", initial_metadata,
+                                std::unique_ptr<GrpcStreamHandlerBase>(
+                                    new MyGrpcStreamHandler())) != WasmResult::ParseFailure) {
+    logError("unexpected bogus service string OK");
+  }
+  if (root()->grpcStreamHandler("cluster", "service", "bad method", initial_metadata,
+                                std::unique_ptr<GrpcStreamHandlerBase>(
+                                    new MyGrpcStreamHandler())) != WasmResult::InternalFailure) {
+    logError("unexpected bogus method OK");
+  }
```

Comment:
Please convert this to a positive test and log that failure occured.

Reply:
done